src/gluonts/dataset/split/splitter.py

```diff
@@ -315,32 +362,36 @@ class DateSplitter(AbstractBaseSplitter, pydantic.BaseModel):
 
     Parameters
     ----------
-    prediction_length
-        Length of the prediction interval in test data.
     split_date
         Period determining where the training data ends. Please make sure
         at least ``prediction_length`` (for ``rolling_split`` multiple of
         ``prediction_length``) values are left over after the ``split_date``.
-    max_history
-        If given, all entries in the *test*-set have a max-length of
-        `max_history`. This can be used to produce smaller file-sizes.
     """
 
-    class Config:
-        arbitrary_types_allowed = True
-
-    prediction_length: int
     split_date: pd.Period
-    max_history: Optional[int] = None
 
     def _train_slice(self, item: TimeSeriesSlice) -> TimeSeriesSlice:
         # the train-slice includes everything up to (including) the split date
         return item[: self.split_date]
 
     def _test_slice(
-        self, item: TimeSeriesSlice, offset: int = 0
-    ) -> TimeSeriesSlice:
-        return item[
-            : self.split_date
-            + (self.prediction_length + offset) * item.start.freq
-        ]
+        self, item: TimeSeriesSlice, prediction_length: int, offset: int = 0
+    ) -> Tuple[TimeSeriesSlice, TimeSeriesSlice]:
+        return (
+            item[: self.split_date],
```

Comment:
I think this should end at
```suggestion
            item[: self.split_date + offset * item.start.freq],
```